client/src/lib/fetch.js

```diff
@@ -0,0 +1,18 @@
+import { SERVER_URL } from "../constants";
+
+export const getUserData = async (id = "") => {
+  const getUserDataEndpoint = `${SERVER_URL}/user/${id}`;
+  const result = await fetch(getUserDataEndpoint, {
+    credentials: "include",
+  });
+  const resJSON = await result.json();
+
+  if (result.status !== 200) {
+    return { ok: false, resJSON };
+  }
+  return { ok: true, resJSON };
```

Comment:
Consider using [`result.ok`](https://fetch.spec.whatwg.org/#ok-status).

```suggestion
  return { resJSON: await result.json(), ok: result.ok };
```